src/diffusers/models/embeddings.py

```diff
@@ -876,9 +876,15 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
 
         x = self.proj_in(x)
 
-        for ln0, ln1, attn, ff in self.layers:
+        for l in self.layers:
             residual = latents
 
+            # make it torch.compile() compatible
+            comps = []
+            for i in l:
+                comps.append(i)
+            [ln0, ln1, attn, ff] = comps
```

Comment:
ok but do we know why it has to be done this way? any documentation etc?
e.g. would this work?
```python
for layer in self.layers:
    [ln0, ln1, attn, ff] = layers
```

Reply:
If you try to unpack a `ModuleList` like above or just like in the original implementation. Following error occurs:
```
torch._dynamo.exc.Unsupported: UNPACK_SEQUENCE NNModuleVariable()
```

So I assume that any unpacking with `ModuleList` is not supported in `torch.compile` and only for loop through it is supported

Reply:
> I also try moving `ln0, ln1, attn, ff` into a class, but it will affect the weight loading

hi @rootonchair, you can use the `IPAdapterPlusImageProjectionBlock` class, but you have to update the loading function too.
I think the final result would be really good, as the code consistency in the library would improve